Lib/test/test_pow.py

```diff
@@ -93,6 +93,28 @@ def test_other(self):
                             pow(int(i),j,k)
                         )
 
+    def test_big_exp(self):
+        import random
+        self.assertEqual(pow(2, 50000), 1 << 50000)
+        # Randomized modular tests, checking the identities
+        #  a**(b1 + b2) == a**b1 * a**b2
+        #  a**(b1 * b2) == (a**b1)**b2
+        prime = 1000000000039 # for speed, relatively small prime modulus
+        for i in range(10):
```

Comment:
Consider modernizing these tests.  Put _a_, _b1_, and _b2_ in a subtest.  Use assertEqual for the _got_ compares.  If there is a failure, the subtest logic with print inputs and the assertEqual logic will print the outputs.

Reply:
These are randomized tests of large ints. Seeing the computed values on failure is useless: you need to know `a`, `b1`, and `b2` to reproduce the failure. I also wanted to force hex output of those, to avoid quadratic-time base conversion for failure output.

Reply:
The hex output is the key then.   Seeing *a*, *b1*, *b2* is what subtest() does for you.

Reply:
OK - I didn't know the `subtest` context manager existed - I avoid unittest whenever conveniently possible :wink:. Instead I fiddled what I wrote to fail on purpose until I could use the failure output to easily reconstruct the failing case. What's there now works fine for that purpose, and it was easy to write it in a way that did so. Is there a point to using the latest elaborate framework feature to do instead what can be easily & obviously done without it? If you were to say you have a hard time _understanding_ the code that's there, then I'd struggle to use the subtest gimmick instead. Else I'd rather not bother.